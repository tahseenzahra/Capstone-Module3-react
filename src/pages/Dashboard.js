/* eslint-disable no-unused-expressions */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import ContinentCard from '../components/ContinentCard';
import { getAllData } from '../redux/world/worldReducer';
import africaImg from '../assets/africa.png';
import europeImg from '../assets/europe.png';
import northImg from '../assets/north-america.png';
import oceaniaImg from '../assets/australia.png';
import southImg from '../assets/south-america.png';
import asiaImg from '../assets/asia.png';

const Dashboard = () => {
  const data = useSelector((state) => state.world);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!data.length) {
      dispatch(getAllData());
    }
  }, []);

  return (
    <div className="dashboard">
      <ContinentCard continents={data} name="Africa" map={africaImg} />
      <ContinentCard continents={data} name="Europe" map={europeImg} />
      <ContinentCard continents={data} name="North-America" map={northImg} />
      <ContinentCard continents={data} name="Oceania" map={oceaniaImg} />
      <ContinentCard continents={data} name="Asia" map={asiaImg} />
      <ContinentCard continents={data} name="South-America" map={southImg} />
    </div>
  );
};

export default Dashboard;
