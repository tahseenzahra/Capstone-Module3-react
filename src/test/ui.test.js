import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from "../redux/configureStore";
import Dashboard from '../pages/Dashboard';

describe('UI Tests', () => {
  test('should render Dashboard', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Dashboard />
        </BrowserRouter>
      </Provider>
    );

    expect(tree).toMatchSnapshot();
  });
});
