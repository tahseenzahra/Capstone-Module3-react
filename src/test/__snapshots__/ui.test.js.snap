// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Tests should render Dashboard 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dashboard"
      >
        <div
          class="continent-card"
        >
          <div
            class="continent-img"
          >
            <img
              alt="Africa Map"
              src="africa.png"
            />
          </div>
          <h2
            class="continent-name"
          >
            Africa
          </h2>
          <p
            class="continent-p"
          >
            <span>
              Total Cases
            </span>
            <span
              style="font-size: 1.2rem; font-weight: bold;"
            >
              0
            </span>
          </p>
          <a
            class="continent-btn"
            href="/continent/Africa"
          >
            View Details
          </a>
        </div>
        <div
          class="continent-card"
        >
          <div
            class="continent-img"
          >
            <img
              alt="Africa Map"
              src="europe.png"
            />
          </div>
          <h2
            class="continent-name"
          >
            Europe
          </h2>
          <p
            class="continent-p"
          >
            <span>
              Total Cases
            </span>
            <span
              style="font-size: 1.2rem; font-weight: bold;"
            >
              0
            </span>
          </p>
          <a
            class="continent-btn"
            href="/continent/Europe"
          >
            View Details
          </a>
        </div>
        <div
          class="continent-card"
        >
          <div
            class="continent-img"
          >
            <img
              alt="Africa Map"
              src="north-america.png"
            />
          </div>
          <h2
            class="continent-name"
          >
            North-America
          </h2>
          <p
            class="continent-p"
          >
            <span>
              Total Cases
            </span>
            <span
              style="font-size: 1.2rem; font-weight: bold;"
            >
              0
            </span>
          </p>
          <a
            class="continent-btn"
            href="/continent/North-America"
          >
            View Details
          </a>
        </div>
        <div
          class="continent-card"
        >
          <div
            class="continent-img"
          >
            <img
              alt="Africa Map"
              src="australia.png"
            />
          </div>
          <h2
            class="continent-name"
          >
            Oceania
          </h2>
          <p
            class="continent-p"
          >
            <span>
              Total Cases
            </span>
            <span
              style="font-size: 1.2rem; font-weight: bold;"
            >
              0
            </span>
          </p>
          <a
            class="continent-btn"
            href="/continent/Oceania"
          >
            View Details
          </a>
        </div>
        <div
          class="continent-card"
        >
          <div
            class="continent-img"
          >
            <img
              alt="Africa Map"
              src="asia.png"
            />
          </div>
          <h2
            class="continent-name"
          >
            Asia
          </h2>
          <p
            class="continent-p"
          >
            <span>
              Total Cases
            </span>
            <span
              style="font-size: 1.2rem; font-weight: bold;"
            >
              0
            </span>
          </p>
          <a
            class="continent-btn"
            href="/continent/Asia"
          >
            View Details
          </a>
        </div>
        <div
          class="continent-card"
        >
          <div
            class="continent-img"
          >
            <img
              alt="Africa Map"
              src="south-america.png"
            />
          </div>
          <h2
            class="continent-name"
          >
            South-America
          </h2>
          <p
            class="continent-p"
          >
            <span>
              Total Cases
            </span>
            <span
              style="font-size: 1.2rem; font-weight: bold;"
            >
              0
            </span>
          </p>
          <a
            class="continent-btn"
            href="/continent/South-America"
          >
            View Details
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dashboard"
    >
      <div
        class="continent-card"
      >
        <div
          class="continent-img"
        >
          <img
            alt="Africa Map"
            src="africa.png"
          />
        </div>
        <h2
          class="continent-name"
        >
          Africa
        </h2>
        <p
          class="continent-p"
        >
          <span>
            Total Cases
          </span>
          <span
            style="font-size: 1.2rem; font-weight: bold;"
          >
            0
          </span>
        </p>
        <a
          class="continent-btn"
          href="/continent/Africa"
        >
          View Details
        </a>
      </div>
      <div
        class="continent-card"
      >
        <div
          class="continent-img"
        >
          <img
            alt="Africa Map"
            src="europe.png"
          />
        </div>
        <h2
          class="continent-name"
        >
          Europe
        </h2>
        <p
          class="continent-p"
        >
          <span>
            Total Cases
          </span>
          <span
            style="font-size: 1.2rem; font-weight: bold;"
          >
            0
          </span>
        </p>
        <a
          class="continent-btn"
          href="/continent/Europe"
        >
          View Details
        </a>
      </div>
      <div
        class="continent-card"
      >
        <div
          class="continent-img"
        >
          <img
            alt="Africa Map"
            src="north-america.png"
          />
        </div>
        <h2
          class="continent-name"
        >
          North-America
        </h2>
        <p
          class="continent-p"
        >
          <span>
            Total Cases
          </span>
          <span
            style="font-size: 1.2rem; font-weight: bold;"
          >
            0
          </span>
        </p>
        <a
          class="continent-btn"
          href="/continent/North-America"
        >
          View Details
        </a>
      </div>
      <div
        class="continent-card"
      >
        <div
          class="continent-img"
        >
          <img
            alt="Africa Map"
            src="australia.png"
          />
        </div>
        <h2
          class="continent-name"
        >
          Oceania
        </h2>
        <p
          class="continent-p"
        >
          <span>
            Total Cases
          </span>
          <span
            style="font-size: 1.2rem; font-weight: bold;"
          >
            0
          </span>
        </p>
        <a
          class="continent-btn"
          href="/continent/Oceania"
        >
          View Details
        </a>
      </div>
      <div
        class="continent-card"
      >
        <div
          class="continent-img"
        >
          <img
            alt="Africa Map"
            src="asia.png"
          />
        </div>
        <h2
          class="continent-name"
        >
          Asia
        </h2>
        <p
          class="continent-p"
        >
          <span>
            Total Cases
          </span>
          <span
            style="font-size: 1.2rem; font-weight: bold;"
          >
            0
          </span>
        </p>
        <a
          class="continent-btn"
          href="/continent/Asia"
        >
          View Details
        </a>
      </div>
      <div
        class="continent-card"
      >
        <div
          class="continent-img"
        >
          <img
            alt="Africa Map"
            src="south-america.png"
          />
        </div>
        <h2
          class="continent-name"
        >
          South-America
        </h2>
        <p
          class="continent-p"
        >
          <span>
            Total Cases
          </span>
          <span
            style="font-size: 1.2rem; font-weight: bold;"
          >
            0
          </span>
        </p>
        <a
          class="continent-btn"
          href="/continent/South-America"
        >
          View Details
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
